spark3/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -90,7 +96,14 @@ public StructType outputType() {
       if (retainLastNum != null) {
         action.retainLast(retainLastNum);
       }
-
+      if (numThreads != null && numThreads > 0) {
+        action.executeDeleteWith(Executors.newFixedThreadPool(numThreads, runnable -> {
+          Thread thread = new Thread(runnable);
+          thread.setName("expire-snapshot-" + deleteThreadsIndex.getAndIncrement());
+          thread.setDaemon(true); // daemon threads will be terminated abruptly when the JVM exits
```

Comment:
if we use exiting executor service, I think this line is not needed? Based on guava doc https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/MoreExecutors.html#getExitingExecutorService(java.util.concurrent.ThreadPoolExecutor):

> Converts the given ThreadPoolExecutor into an ExecutorService that exits when the application is complete. It does so by using daemon threads and adding a shutdown hook to wait for their completion.